test/zdtm.py

```diff
@@ -626,14 +626,14 @@ def available():
                 ["make", "zdtm_ct"], env=dict(os.environ, MAKEFLAGS=""))
         if not os.access("zdtm/lib/libzdtmtst.a", os.F_OK):
             subprocess.check_call(["make", "-C", "zdtm/"])
-        if opts['rootless']:
+        if 'rootless' in opts and opts['rootless']:
```

Comment:
I think we should not run these actions if we are not going to execute tests:
diff --git a/test/zdtm.py b/test/zdtm.py
index a311610c3..99b59f551 100755
--- a/test/zdtm.py
+++ b/test/zdtm.py
@@ -618,6 +618,8 @@ class zdtm_test:
 
     @staticmethod
     def available():
+        if opts['action'] != "run":
+            return
         if not os.access("umount2", os.X_OK):
             subprocess.check_call(
                 ["make", "umount2"], env=dict(os.environ, MAKEFLAGS=""))
@@ -633,6 +635,8 @@ class zdtm_test:
 
     @staticmethod
     def cleanup():
+        if opts['action'] != "run":
+            return
         if opts['rootless']:
             return
         subprocess.check_call(

Reply:
It is slightly more complicated. `available()` invokes `make -C zdtm/` that then call `make` in the `lib`, `static`, and `transition` sub-directories to build the tests. Then, the `list` command counts and prints the list of tests that have been built. Thus, with this change `./zdtm.py list` would not work.

Reply:
> available() invokes make -C zdtm/ that then calls make in the lib, static, and transition sub-directories to build the tests.

This part never works as expected, because it builds tests only if zdtm/lib/libzdtmtst.a doesn't exist, but it doesn't mean that all tests are compiled and binaries are up to date. I don't think that zdtm.py has to compile anything.